python/mlc_chat/cli/gen_config.py

```diff
@@ -76,6 +76,12 @@ def _parse_output(path: Union[str, Path]) -> Path:
         default=None,
         help=HELP["tensor_parallel_shards"] + ' (default: "%(default)s")',
     )
+    parser.add_argument(
+        "--max-batch-size",
+        type=int,
+        default=80,
```

Comment:
curious why 80 here?

Reply:
what if the batch size exceeds 80 when #requests surge?

Reply:
I think this would results in queuing of pending requests. This is max batch that can be processed per single step batching

Reply:
80 is the roughly the batch size upper bound in our previous pressure test for serving 7B llama model. I think it could default to 64, and it can be adjusted anyways.

The `max_batch_size` here is set for memory planning, which means if the vm prefill/decode function takes batch size larger than specified `max_batch_size`, it could lead to error (similar to taking a sequence longer than `context_window_size`). On the other hand, the serving engine reads the config JSON as well and ensures the concurrent running requests does not exceed this number.

Reply:
Got it. Thanks for the explanation!